src/rgw/rgw_rest_swift.cc

```diff
@@ -498,13 +498,13 @@ static int get_swift_container_settings(req_state * const s,
 
   if (read_attr || write_attr) {
     RGWAccessControlPolicy_SWIFT swift_policy(s->cct);
-    int r = swift_policy.create(store,
+    bool r = swift_policy.create(store,
                                 s->user->user_id,
                                 s->user->display_name,
                                 read_list,
                                 write_list);
-    if (r < 0) {
-      return r;
+    if (r != true) {
+      return -EINVAL;
```

Comment:
`-EINVAL` should be mapped to `400 Bad Request` which is completely OK according to what Swift is doing:

```
 swift$ curl -i -H "X-Auth-Token: $token" "$publicURL" -X POST -H "X-Account-Access-Control: bla"
HTTP/1.1 400 Bad Request
Content-Length: 76
Content-Type: text/plain; charset=UTF-8
X-Trans-Id: txe731606404814e4fbefab-0057b71d87
Date: Fri, 19 Aug 2016 14:53:59 GMT

X-Account-Access-Control invalid: Syntax error in input ('bla')

Input: bla
```